engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/ProcessInstanceStateTransitionGuard.java

```diff
@@ -37,15 +39,19 @@ public ProcessInstanceStateTransitionGuard(final BpmnStateBehavior stateBehavior
    *
    * @return {@code true} if the preconditions are met and the transition command is valid.
    */
-  public Either<Failure, ?> isValidStateTransition(final BpmnElementContext context) {
-    return checkStateTransition(context).mapLeft(Failure::new);
+  public Either<Failure, ?> isValidStateTransition(
+      final BpmnElementContext context, final ExecutableFlowElement element) {
+    return checkStateTransition(context, element).mapLeft(Failure::new);
   }
 
-  private Either<String, ?> checkStateTransition(final BpmnElementContext context) {
+  private Either<String, ?> checkStateTransition(
+      final BpmnElementContext context, final ExecutableFlowElement element) {
     switch (context.getIntent()) {
       case ACTIVATE_ELEMENT:
-        return hasActiveFlowScopeInstance(context);
-
+        final Either<String, ?> hasActiveFlowScopeInstance = hasActiveFlowScopeInstance(context);
+        return hasActiveFlowScopeInstance.isLeft()
+            ? hasActiveFlowScopeInstance
+            : canActivateParallelGateway(context, element);
```

Comment:
🔧 Just a quick tip: You can use `flatMap(ok -> canActivateParallelGateway())` to achieve the same as the `if(failed) ? return early : else do another check that can fail and return that`